src/shared_modules/content_manager/src/components/factoryDownloader.hpp

```diff
@@ -30,14 +34,34 @@
  */
 class FactoryDownloader final
 {
+private:
+    /**
+     * @brief Deduces and returns the compression type given an input file extension.
+     *
+     * @param inputFile Input file whose compression type will be deduced.
+     * @return std::string Compression type.
+     */
+    static std::string deduceCompressionType(const std::string& inputFile)
+    {
+        const std::map<std::string, std::string> COMPRESSED_EXTENSIONS {{".gz", "gzip"}, {".xz", "xz"}};
+        const auto& fileExtension {std::filesystem::path(inputFile).extension()};
+
+        if (COMPRESSED_EXTENSIONS.find(fileExtension) != COMPRESSED_EXTENSIONS.end())
+        {
+            return COMPRESSED_EXTENSIONS.at(fileExtension);
+        }
```

Comment:
since the `find` method returns an iterator, we could reuse it for the return
```suggestion
        auto compressedExtension {COMPRESSED_EXTENSIONS.find(fileExtension)};

        if (compressedExtension != COMPRESSED_EXTENSIONS.end())
        {
            return compressedExtension->second;
        }
```

Reply:
Done!